### ISSUE TITLE
A new library (JxInsta) is now out for the testing. Check it out!

### ISSUE THREAD

**Issue by ErrorxCode** [instagram4j/instagram4j#719]
<p align="center"><img alt="JxInsta" height="300" src="https://github.com/ErrorxCode/JxInsta/blob/main/JxInsta.png?raw=true"></p>
<p align="center">
  <img src="https://img.shields.io/github/license/ErrorxCode/JxInsta?style=for-the-badge">
  <img src="https://img.shields.io/github/stars/ErrorxCode/JxInsta?style=for-the-badge">
  <img src="https://img.shields.io/github/issues/ErrorxCode/JxInsta?color=red&style=for-the-badge">
  <img src="https://img.shields.io/github/forks/ErrorxCode/JxInsta?color=teal&style=for-the-badge">
</p>

<p align="center">
  <img src="https://img.shields.io/badge/Author-Rahil-cyan?style=flat-square">
  <img src="https://img.shields.io/badge/Open%20Source-Yes-cyan?style=flat-square">
  <img src="https://img.shields.io/badge/Written%20In-Java-cyan?style=flat-square">
</p>

# [JxInsta](https://github.com/ErrorxCode/JxInsta)
A java library of Instagram private web API (*may include mobile api in future*).  This library is built as a replacement for **instagram4j** since it was too old fashioned and was facing a challenge problem due to a similar header population problem among it's users. This project is lead by **[EasyInsta](https://github.com/ErrorxCode/EasyInsta)** and is part of it.

## Features

-   Lightweight and Easy 2 use, Object-oriented
-   No need API token
-   Supports  **Sending messages**
-   Supports  **Getting/fetching messages**
-   Supports  **Deleting message**
-   Supports  **_Realtime direct messages listener_** (Pending)
-   Supports  **Login using cache/saving sessions**
-   Supports  **Posting (Picture)**
-   Supports  **Adding stories (Photo)**
-   Supports  **Following/Unfollowing**
-   Supports  ***Acception/Ignoring follow request*** (Pending)
-   Supports  **Scrapping followings and followers**
-   Supports  **Getting profile data**
-   Supports  **Liking/commenting on post**
-   Supports  **Fetching feeds/timeline post**
-   Supports  **Downloading posts and pfp**

**Comment by DavidTheExplorer** [comment]
Did the repository's owner authorize you to declare the continuation of the library?

**Comment by ErrorxCode** [comment]
Nope, as part of this community, I have taken the responsibility to continue the development of **Instargam private API**.
Also, the project is open-source, So I do not require any authorization from the owner

**Comment by DavidTheExplorer** [comment]
The problem is **not** that you built your own API, it's that you're declaring the continuation of a project without permission - which is misleading.
We already trust _Instagram4J_ - how do we know your project is not a virus? just because you said it's a continuation?

@jvogit I know you're inactive but please check this out.

**Comment by ErrorxCode** [comment]
LoL. First of all, viruses are never published like this. Secondly, The project is open-source, you can check every line of the code if you doubt it.

I am doing this to let users of **Instagram4j** know about the new library (because they are struggling with the no-response issue). They have already been using **EasyInsta** which is a wrapper of this library. The **JxInsta** is a part of it

**Comment by DavidTheExplorer** [comment]
Why are you not addressing the problem I'm raising? Even if you have a good will, that doesn't allow you to say that you're continuing a project - You have to **clearly** mention the separation of the two projects.
What I said about a potential virus is just an example - If someone really thinks your project is a continuation - they most likely wouldn't even check the source code.
I'm not against you here, I simply don't want people to be misled.

**Comment by syedhilalhusain** [comment]
> Why are you not addressing the problem I'm raising? Even if you have a good will, that doesn't allow you to say that you're continuing a project - You have to **clearly** mention the separation of the two projects.
> What I said about a potential virus is just an example - If someone really thinks your project is a continuation - they most likely wouldn't even check the source code.
> I'm not against you here, I simply don't want people to be misled.

Bro I think you should mind your own business,  you're not responsible for everyone on github, people are not as dumb as you think, ofc they'll check for any potential viruses. Also this is an open source library and I use it, and it works perfectly fine.
PS : every discovery needs enhancements or improvement over the time for better functionality and the developer of Instagram4j isn't active from a millenium, he didn't add any new feature to the Library let alone improve it, so obviously the user's need a better version.

**Comment by jvogit** [comment]
I appreciate the work @ErrorxCode has done and anyone is free to check it out as it's open source anyone can do look through and use it. I don't have any issue with another developed library, using references from this library, or creating a fork as its free software 👍 

I appreciate the concern @DavidTheExplorer brought up. **I've not endorsed a new library over instagram4j and this is a completely separate _independent_ project**. But I am no longer actively developing this library due to extremely limited time. It's very hard to work on this type of library because you have to constantly keep up to date with API requests and at any moment Meta can nuke it off GitHub.

That said I've locked it as I think there's not much more meaningful discussion. Thanks!
